misk/src/main/kotlin/misk/web/actions/ReadinessCheckAction.kt

```diff
@@ -1,31 +1,56 @@
 package misk.web.actions
 
 import com.google.common.util.concurrent.ServiceManager
+import com.google.common.util.concurrent.ThreadFactoryBuilder
 import misk.healthchecks.HealthCheck
 import misk.security.authz.Unauthenticated
 import misk.web.AvailableWhenDegraded
 import misk.web.Get
 import misk.web.Response
 import misk.web.ResponseContentType
+import misk.web.WebConfig
 import misk.web.mediatype.MediaTypes
 import wisp.logging.getLogger
 import javax.inject.Inject
 import javax.inject.Provider
 import javax.inject.Singleton
+import kotlinx.coroutines.*
+import misk.healthchecks.HealthStatus
+import java.time.Clock
+import java.time.Instant
+import java.util.concurrent.SynchronousQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
 
 private val logger = getLogger<ReadinessCheckAction>()
 
 @Singleton
 class ReadinessCheckAction @Inject internal constructor(
   private val serviceManagerProvider: Provider<ServiceManager>,
-  @JvmSuppressWildcards private val healthChecks: List<HealthCheck>
+  @JvmSuppressWildcards private val healthChecks: List<HealthCheck>,
+  val config: WebConfig,
+  val clock: Clock
 ) : WebAction {
 
+  @Volatile
```

Comment:
Do we need `Volatile`, since the thread pool is bound to a single thread?

Reply:
The read comes from a different thread than the write, since its not on the refreshPool

Reply:
Even if we're updating status only from one thread, we're going to be reading it from another. So, we need to consider the memory visibility of our cached status. `@Volatile` seems adequate for the current structure but we could consider alternate concurrency primitives if desired.